hack/verify-staticcheck.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+kube::golang::verify_go_version
+
+FOCUS="${1:-}"
+
+# TODO: Start verifying more checks!
+# See https://staticcheck.io/docs/checks
+CHECKS=(
+  "U1*" # Detect unused code
```

Comment:
these also seem useful, did you encounter issues with them?:
* SA4006 | A value assigned to a variable is never read before being overwritten. Forgotten error check or dead code?
* SA4017 | A pure function's return value is discarded, making the call pointless
* SA4010 | The result of append will never be observed anywhere
* SA4020 | Unreachable case clause in a type switch
* SA5??? | Correctness issues

once we have the package blacklist and start resolving them, it will be hard to add in additional checks

Reply:
I was thinking we could tackle 1 at a time, but I'm open to adding more. I also considered a blacklist of failure instances, rather than packages, but was worried it would cause too much churn.

Reply:
> once we have the package blacklist and start resolving them, it will be hard to add in additional checks

Yup. It's better to start big (like ~all checks) and then specifically ignore things as you go through package by package: https://staticcheck.io/docs/#ignoring-problems

Reply:
Looking at the list a bit more closely, it seems like "S1??? code simplifications" and "ST1???" are more stylistic. Not necessarily wrong, but mostly unnecessary.

So maybe we want: `all,-S1*,-ST1*`?

Reply:
I'm +1 on enabling all checks for actual problems, and omitting stylistic checks